src/Scriban/Functions/ArrayFunctions.cs

```diff
@@ -182,30 +180,59 @@ public static object Cycle(TemplateContext context, SourceSpan span, IList list,
         /// </remarks>
         public static ScriptRange Each(TemplateContext context, SourceSpan span, IEnumerable list, object function)
         {
-            if (list == null) return null;
-            if (function == null) return new ScriptRange(list);
-            
-            var scriptingFunction = function as IScriptCustomFunction;
-            if (scriptingFunction == null)
+            return ApplyFunction(context, span, list, function, EachProcessor);
+        }
+        private static IEnumerable EachInternal(TemplateContext context, SourceSpan span, IEnumerable list, IScriptCustomFunction function, Type destType)
+        {
+            var arg = new ScriptArray(1);
+            foreach (var item in list)
             {
-                throw new ArgumentException($"The parameter `{function}` is not a function. Maybe prefix it with @?", nameof(function));
+                var itemToTransform = context.ToObject(span, item, destType);
+                arg[0] = itemToTransform;
+                var itemTransformed = ScriptFunctionCall.Call(context, context.CurrentNode, function, arg);
+                yield return itemTransformed;
             }
+        }
+
+        static readonly ListProcessor EachProcessor = EachInternal;
 
-            return new ScriptRange(EachInternal(context, span, list, scriptingFunction, scriptingFunction.GetParameterInfo(0).ParameterType));
+        /// <summary>
+        /// Filters the input list according the supplied filter function.
+        /// </summary>
+        /// <param name="context">The template context</param>
+        /// <param name="span">The source span</param>
+        /// <param name="list">An input list</param>
+        /// <param name="function">The function used to test each elemement of the list</param>
+        /// <returns>Returns a new list which contains only those elements which match the filter function.</returns>
+        /// <remarks>
+        /// ```scriban-html
+        /// {{ [1, 200, 3,400] | array.each @(do;ret $0>=100 ; end)}}
```

Comment:
```suggestion
        /// {{ [1, 200, 3,400] | array.filter @(do;ret $0>=100 ; end)}}
```

The test is failing after generating the doc. It should be `array.filter` instead